bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromBulbHandler.java

```diff
@@ -0,0 +1,314 @@
+                updateState(CHANNEL_MODE, StringType.valueOf(deviceInfo.mode));
+            }
+            updateState(CHANNEL_RAMP, QuantityType.valueOf(deviceInfo.ramp, MetricPrefix.MILLI(SECOND)));
+            if (deviceInfo instanceof MyStromDeviceSpecificInfo) {
+                updateState(CHANNEL_POWER, QuantityType.valueOf(((MyStromDeviceSpecificInfo) deviceInfo).power, WATT));
+            }
+        }
+    }
+
+    /**
+     * Given a URL, send a HTTP GET request to the URL location
+     * created by the URL.
+     *
+     * @return String contents of the response for the GET request.
+     * @throws MyStromException Throws on communication error
+     */
+    private String getDeviceSpecificInformation() throws MyStromException {
+        String url = getHostname() + "/api/v1/device";
+        ContentResponse response;
+        try {
+            Request request = getHttpClient().newRequest(url).timeout(10, TimeUnit.SECONDS).method(HttpMethod.GET);
+            response = request.send();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new MyStromException(COMMUNICATION_ERROR + e.getMessage());
+        }
+
+        if (response.getStatus() != HTTP_OK_CODE) {
+            throw new MyStromException(
+                    "Error sending HTTP GET request to " + url + ". Got response code: " + response.getStatus());
+        }
+        return response.getContentAsString();
+    }
+
+    /**
+     * Given a URL and a set parameters, send a HTTP POST request to the URL location
+     * created by the URL and parameters.
+     *
+     * @param action The action we want to take (on,off or toggle)
+     * @param color The color we set the bulb to (When using RGBW mode the first two hex numbers are used for the
+     *            white channel! hsv is of form <UINT 0..360>;<UINT 0..100>;<UINT 0..100>)
+     * @param ramp Transition time from the light’s current state to the new state. [ms]
+     * @param mode The color mode we want the Bulb to set to (rgb or hsv or mono)
+     * @return String contents of the response for the GET request.
+     * @throws MyStromException Throws on communication error
+     */
+    private String sendHttpPost(@Nullable String action, @Nullable String color, @Nullable String ramp,
+            @Nullable String mode) throws MyStromException {
+        String url = getHostname() + "/api/v1/device/" + getMac();
+        ContentResponse response;
+        try {
+            Fields fields = new Fields();
+            if (action != null) {
+                fields.put("action", action);
+            }
+            if (color != null) {
+                fields.put("color", color);
+            }
+            if (ramp != null) {
+                fields.put("ramp", ramp);
+            }
+            if (mode != null) {
+                fields.put("mode", mode);
+            }
+
+            // FormContentProvider contentProvider = new FormContentProvider(fields);
```

Comment:
What's wrong with using the `FormContentProvider`? I would prefer that ove rthe hard-to-understand-code below. I

Reply:
Ok. I will test again. I meant I couldn't get it to work with that, because the hsv String is separated with a semicolon ;

Reply:
Test again. Doesn't work. The `FormContentProvider` does `encode(...)`. If it's encoded, it doesn't work.

Reply:
https://github.com/eclipse/jetty.project/blob/1701b83fd5035cbd9ab88dd7f3b210190776a6c0/jetty-client/src/main/java/org/eclipse/jetty/client/util/FormContentProvider.java#L53-L63 is a better (easier to understand) implementation. And please remove the FormContentProvider(no commented code).

After that please push your changes, the PR will automatically be updated. Do not use force-push, otherwise the changes are not visible and harder to review.